DEVELOPER.md

```diff
@@ -0,0 +1,43 @@
+# Notes for Development
+
+## Development Environment
+
+## Building from Source
+
+To build and test the driver, the standard cargo commands can be used from the root directory.
+
+For an unoptimized build with debugging information (most common):
+```
+cargo build
+```
+
+For an optimized build with debugging info (mac)
+```
+cargo build --features odbc-sys/iodbc,cstr/utf32 --profile=release-with-debug
+```
+And for an optimized build with debugging info (windows, linux)
+```
+cargo build --profile=release-with-debug
+```
+The resulting build files will be output to the `target` directory in the `debug` or `release` folders respectively.
+
+## Running Tests
+
+### To run unit tests
+
+Similar to building, standard cargo commands can be used here:
+
+```
+cargo test unit
+```
+
+### Other types of tests
```

Comment:
note: I considered transcribing steps from evergreen.yml here, but given that would require updating this doc anytime they update (which seems not totally uncommon), I left it like this for now, but would reconsider if it may be worthwhile